### PR TITLE
remove functional style collection processing

### PR DESCRIPTION
**Pull request by NamespaceV**

### CODE REVIEW

File: Scripts/Battle/battle_manager.gd
Comment (Wiolarz) [inline Scripts/Battle/battle_manager.gd:1]: isn't this better:
if is_legal_summon_coord(c, player):
    result.append(B_GRID.get_tile(c))

File: Scripts/UI/battle_ui.gd
Comment (Wiolarz) [inline Scripts/UI/battle_ui.gd:1]: Isn't this better?
for army in armies:
----if army.controller == player:
---- ----current_armies.append(army)

File: Scripts/Battle/battle_manager.gd
Comment (NamespaceV) [reply Scripts/Battle/battle_manager.gd:1]: changed

File: Scripts/UI/battle_ui.gd
Comment (NamespaceV) [reply Scripts/UI/battle_ui.gd:1]: changed

**Pull request merged**